server/pulp/server/async/scheduler.py

```diff
@@ -121,30 +113,39 @@ def check_celery_processes(self):
         present. If there are zero of either, log at the error level that Pulp will not operate
         correctly.
         """
-        msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager '
-                'processes are missing for more than %d seconds') % self.CELERY_TIMEOUT_SECONDS
+        msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager processes '
+                'are missing for more than %d seconds') % constants.CELERY_TIMEOUT_SECONDS
         _logger.debug(msg)
         now = ensure_tz(datetime.utcnow())
-        oldest_heartbeat_time = now - timedelta(seconds=self.CELERY_TIMEOUT_SECONDS)
+        oldest_heartbeat_time = now - timedelta(seconds=constants.CELERY_TIMEOUT_SECONDS)
         worker_list = Worker.objects.all()
         worker_count = 0
         resource_manager_count = 0
         scheduler_count = 0
+
         for worker in worker_list:
             if worker.last_heartbeat < oldest_heartbeat_time:
                 msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
                 _logger.error(msg)
-                _delete_worker(worker.name)
-            elif worker.name.startswith(SCHEDULER_WORKER_NAME):
+
+                # Check if worker is a scheduler, and clean up appropriately
```

Comment:
Years ago a friend of mine convinced me not to put comments of intent like this because over time they don't get maintained and then they become wrong. Instead he encouraged me to write "crazy readable code" which was as precise as code but as readable as text. This is like an implicit comment.

For example in this case, `worker.delete()` could be renamed to `worker.delete_db_entry()` and `_delete_worker(worker.name)` could be `_delete_worker_db_cancel_tasks_cleanup_reservations(worker.name)`

You don't necessarily have to do all this, I'm just giving an example about readable code.